### PR TITLE
Bug 1134890 upgrade to django 15

### PR DESCRIPTION
**Pull request by peterbe**

### CODE REVIEW

**Comment by peterbe** [comment]
@camd any idea why travis doesn't run pull requests?

**Comment by peterbe** [comment]
Hey @shinglyu and @camd 
This monster of a PR is going to be hard to review but I want you to be aware of it. We can't not go ahead with this. We need to upgrade Django to a supported version because old versions don't get security fixes. 

This whole PR is about 1 original thing: Upgrade django from 1.4 to 1.5. 

Doing so required me to also upgrade django-webtest and django-tastypie. 
And django-tastypie has changed **a lot** from 0.9.11 to 0.12.1. To begin with, the whole permission model has changed. Another big change is that IDs, that are integers, used to be returned as strings (e.g `"123"`) but now they're returned as ints (e.g. `123`). 

I guess it's going to be hard to review this. Perhaps we just land it, stick it on dev and bang at it till it works. 

My plan is to upgrade a bunch of other dependencies too. Like django-browserid and django-compressor. However, the next goal is to continue to upgrade Django to 1.6. We can't go all the way to Django 1.7.

**Comment by camd** [comment]
Peter: Not sure if it’s worth it, but we could split the commits to make it more review-able.  Like with ``git add -p``.  Though I’ve been using Atlassian SourceTree to do it and that is MUCH easier.  You can split chunks of changes up and makes it much easier to follow.

It may not be worth it in this case.  Sounds like the biggest issue was the permissions thing, though.  And I think I understand that problem as well as I’m going to…  :)  So no need to review for that.

So maybe, like you say, we could just go with it.

Fwiw- I skipped 1.6 and went from 1.5 to 1.7 and that wasn’t that bad.  A few issues wrt migrations, which was the big difference.  But the python 2.6/2.7 issue may be what stops us on that.

Maybe I should take a quick look through it to see if anything stands out.  Did you want to merge that today?
-Cam

**Comment by peterbe** [comment]
> Maybe I should take a quick look through it to see if anything stands out.  Did you want to merge that today?

Not today. But some time early next week would be nice. 
Don't scrutinize. That would take too long for you. But skim through for big obvious problems. 

One thing that makes me a tiny percent worried and something I just opted to not worry about for now is that in Django 1.5 there's a validator regex on the username thus now disallowing spaces or `:` characters and we have some of those in the mysql database. However, this will go away over time since we'll be using Persona which generates a safe-looking username. 

It *might* matter if you do some editing of old user accounts maybe.

File: moztrap/model/mtapi.py [hunk 1]
```diff
@@ -61,10 +63,50 @@ def is_authenticated(self, request, **kwargs):
         return self.get_key(user, api_key)
 
 
+# # # Debuggin'
```
Comment (camd) [inline moztrap/model/mtapi.py]: should we remove this now?  or still useful for debuggin'...  :)

File: moztrap/model/mtapi.py [hunk 2]
```diff
@@ -75,17 +117,52 @@ def permission(self):
...
+        return super(MTAuthorization, self).update_detail(object_list, bundle)
+
+    def delete_detail(self, object_list, bundle):
+        klass = self.base_checks(bundle.request, bundle.obj.__class__)
+        if klass and bundle.request.user.has_perm(self.permission):
+            return True
+        return super(MTAuthorization, self).delete_detail(object_list, bundle)
+
+
+    # def is_authorized(self, request, object=None):
```
Comment (camd) [inline moztrap/model/mtapi.py]: looks like we can remove this, too?

File: moztrap/model/mtapi.py [hunk 3]
```diff
@@ -103,6 +180,7 @@ class Meta:
         detail_allowed_methods = ["get", "put", "delete"]
         authentication = MTApiKeyAuthentication()
         authorization = MTAuthorization()
+        # authorization = DjangoAuthorization()
```
Comment (camd) [inline moztrap/model/mtapi.py]: remove

File: moztrap/view/manage/cases/forms.py [hunk 4]
```diff
@@ -109,7 +109,14 @@ def is_valid(self):
     def save_attachments(self, caseversion):
         # @@@ convert into a modelmultiplechoicefield widget?
         delete_ids = set(self.data.getlist("remove-attachment"))
-        caseversion.attachments.filter(id__in=delete_ids).delete()
+
+        # TODO!
```
Comment (camd) [inline moztrap/view/manage/cases/forms.py]: should we enter a bug for this?  not sure.  maybe not necessary if the functionality is still working ok.

**Comment by peterbe** [comment]
You merged it. I didn't get a chance to delete all the left-over debugging that was commented out or left defunct. Shall I clean those up after?

**Pull request merged**